### ISSUE TITLE
Data format definition

### ISSUE THREAD

**Issue by ljm565** [ljm565/universal-llm-trainer#4]
### 현재 문제점
```json
{
   "alpaca_style":{
      "train":[
         {
            "input": [],
            "instruction": ["Please recommend summer vacation spots.", "I like mountains."],
            "output": ["In summer, I feel like I could choose between the mountains and the sea.", "If you like mountains, how about the Dolomites?"]
         },
         ...
      ],
      "validation":[
         {
            "input": ["You are an language translator."],
            "instruction": ["Please translate the below sentence in English.\n\n나는 너를 사랑해."],
            "output": ["Here is the results: I love you."]
         },
         ...
      ]
   }
}
```
 1. 용어 변경 필요 (완료)
      * ~~input -> system_prompt~~
      * ~~instruction -> user_prompt~~
      * ~~output -> response~~
 2. train/validation이 하나의 데이터에 묶여있다는 부분 (완료)
       * ~~분리 필요~~
 3. pickle 파일로 되어있다는 부분 (완료)
      * ~~json 형식으로 통일할 필요가 있음.~~
 4. Value type 결정 (완료) 
       * ~~input, instruction, output은 list인데, 현상 유지 하는 것으로 결정.~~

### 바뀌어야하는 코드
1. `AutoregressiveDataset`, `SFTDataset`에서 사용되는 용어들.
2. ~~`src/univlt/trainer/build.py`의 `build_llm_dataset`에서 training mode에 따라 dataset 불러오는 부분 수정. (train mode에서 validation.json 존재 시에만 validation set 사용, 그 외에는 train set만 사용)~~ @Quasar0311 

<br><br>

### 현재 문제점
1. 용어 변경 필요
     * ~~prompt_input -> system_prompt_template~~
     * ~~prompt_no_input -> no_system_prompt_template  --> 없애고 위에걸로 통일 후, system prompt가 없을 때, {input} 부분을 공백으로 치환.~~
     * ~~input, instruction 키워드들 변경~~
2. 형식 변경
     * ~~1번에 해당하는 key의 value 타입이 list인데, str로 변경 필요.~~
3. ~~`config/*.yaml` 파일에 `template_dir`이 list[str]인데 str로 수정 필요.~~
4. ~~`src/univlt/config/dataset_config.py`의 `template_dir: list[str]`를 `template_dir: str`로 수정.~~
5. ~~`template_dir` 대신 `template_path`로 `config`에 사용, `templates/` 디렉토리 내에 모델 별 분할 대신 template 파일에 모델명 달아주는 형태로 수정.~~

### 바뀌어야하는 코드
1. ~~`AutoregressiveDataset`의 self.templates, `random.choice`로 템플릿 랜덤 선택하는 부분 -> str 형식으로 변경하면 하나로 정해지므로 불필요~~
2. ~~`SFTDataset`의 self.templates, `random.choice`로 템플릿 랜덤 선택하는 부분 -> str 형식으로 변경하면 하나로 정해지므로 불필요~~
3. ~~`build_llm_dataset` 함수~~
    * ~~`dataset_paths` pkl이 아닌 json으로 로드 할 수 있게끔 수정.~~
    * ~~`if not cfg.data_cfg.template_dir:  /   else` 문도 수정.~~

<br><br>

### ~~AR 학습 방식~~ @Quasar0311 
```
pad_token_id == eos_token_id == 2 일 때,

* input_ids
A: [10, 11, 12, 13, 14, 2]
B: [20, 21, 22, 2, 2, 2] ← (여기서 뒤의 2들은 padding 역할)

* attention_mask
A: [1, 1, 1, 1, 1, 1] ← 마지막 eos는 “진짜 eos”
B: [1, 1, 1, 1, 0, 0] ← 뒤의 eos들은 “padding eos”

*labels (padding의 역할을 하는 eos 제외)
A: [..., 14, 2]
B: [..., 2, -100, -100]

* 맨 앞 <bos> 토큰은 제외한 후 학습 데이터 생성.
* 주의 `return self.tokenizer.encode(s, add_special_tokens=False)`
```

### ~~SFT 학습 방식~~ @ljm565 
```
* Loss gradient 주는 부분
system ❌
user ❌
assistant (turn1) ✅ 
user ❌
assistant (turn2) ✅

* 맨 앞 모델 별 사용된 <bos> 토큰이 있으면 추가한 후 후 학습 데이터 생성.
* 주의 `return self.tokenizer.encode(s, add_special_tokens=False)`
```
